packages/redot-stringify/index.js

```diff
@@ -0,0 +1,25 @@
+import {stringify} from 'ts-graphviz/ast'
+
+/**
+ * @typedef {(Node extends { type: 'Dot' } ? Node : never)} OnlyDotNode
+ * @template Node
+ */
+
+/**
+ * @typedef {OnlyDotNode<Parameters<typeof stringify>[0]>} Dot
```

Comment:
@kamiazya not sure if this is intentional or not, the AST node TypeScript definitions don't appear to be exported.
I ended up doing a work around to extract them from the `stringify` and `parse` functions indirectly.

Reply:
@ChristianMurphy It should be exported from the ts-graphviz/ast module as `ASTNode` or `DotASTNode`, but can you try to import it?

For this use case, DotASTNode would be appropriate.

Please also refer to the sample I created for verification in Stackblitz.

https://stackblitz.com/edit/ts-graphviz-ast-types?file=index.ts

Reply:
Thanks @kamiazya, somehow I missed it was scoped under the `ast` submodule.
I've updated it to use the type directly, thanks again!